### PR TITLE
[React]: Add initial usage of the new web worker-based cody web chat

### PR DESCRIPTION
**Pull request by vovakulikov**
Part of https://github.com/sourcegraph/sourcegraph/issues/62448
Linear issue [SRCH-573](https://linear.app/sourcegraph/issue/SRCH-573/integrate-cody-web-package-into-the-sourcegraph-ui) 

This is highly experimental usage of the new (not currently merged but published in NPM `cody-web-experimental`) package

## How to run it 
- (Optional) if you previously linked any local packages make sure they don't exist in your node_modules anymore, `rm -rf node_modules` in the root then `pnpm install`
- Run standard `sg start web-standalone`
- Turn on `newCodyWeb: true` in your `experimentalFeatures`

## How to run it locally with prototype PR in Cody repository 

- Open Cody repository on the `vk/integrate-cody-web-chat-2` branch 
- At the root of the repo, run `pnpm install` to make sure you're up to date with all of the dependencies.
- Go to the web package (`cd web`)
- Build it with `pnpm build`
- Create a global link with `pnpm link --global` (Ignore the warning message about no binary)
- Open sourcegraph/sourcegraph repository on this PR branch
- Make sure you are in the root of the repo.
- Run `pnpm link --global cody-web-experimental`
- Run `sg start web-standalone` to bundle the web app and launch an instance that uses S2 for the backend. You'll need to create a login on S2 that is not federated by GitHub.
- Turn on `newCodyWeb: true` in your `experimentalFeatures`
- Have fun experimenting!

## Test plan
- Check that old version of Cody has got no regressions

### CODE REVIEW

File: client/web/bundlesize.config.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const config = {
     },
     {
       path: path.join(STATIC_ASSETS_PATH, 'chunks/chunk-*.js'),
-      maxSize: '600kb', // 2 monaco chunks are very big
+      maxSize: '6.45MB', // Cody web chunk is very big
```
Comment (vovakulikov) [inline client/web/bundlesize.config.js:25]: Unfortunately, I couldn't find a way to name Cody Web chunk with a specific name. 
We've already been violating this rule in different ways so I think this is tolerable

File: client/web/src/cody/chat/old-chat/CodyChatPage.tsx [hunk 2]
```diff
@@ -0,0 +1,419 @@
+import React, { useEffect, useState } from 'react'
```
Comment (vovakulikov) [inline client/web/src/cody/chat/old-chat/CodyChatPage.tsx:1]: This file just a copy of the old page, no logic has been changed here

**Review by fkling — APPROVED** [review]
I didn't do a detail review but I looked through it. Since it's gated behind a flag it seems safe to merge. Just want to point out https://github.com/sourcegraph/sourcegraph/pull/63430 which also makes changes to `CodyChatPage.tsx`, so that needs to be coordinated somehow.

**Review by peterguy — APPROVED** [review]
Ran a functional test locally against S2; it behaved correctly. Didn't try to break it too much.

**Pull request merged**